src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -172,82 +176,97 @@ private function getPendingMessages(): \Generator
-                'MessageAttributeName.1' => 'All',
+                'MessageAttributeNames' => ['All'],
                 'WaitTimeSeconds' => $this->configuration['wait_time'],
             ]);
         }
 
-        if ($this->client->stream($this->currentResponse, $this->configuration['poll_timeout'])->current()->isTimeout()) {
+        if (!$this->fetchMessage()) {
             return;
         }
 
-        $xml = new \SimpleXMLElement($this->currentResponse->getContent());
-        foreach ($xml->ReceiveMessageResult->Message as $xmlMessage) {
+        yield from $this->getPendingMessages();
+    }
+
+    private function fetchMessage(): bool
+    {
+        if (!$this->currentResponse->resolve($this->configuration['poll_timeout'])) {
+            return false;
+        }
+
+        foreach ($this->currentResponse->getMessages() as $message) {
             $headers = [];
-            foreach ($xmlMessage->MessageAttribute as $item) {
-                if ('String' !== (string) $item->Value->DataType) {
+            foreach ($message->getMessageAttributes() as $name => $attribute) {
+                if ('String' !== $attribute->getDataType()) {
                     continue;
                 }
-                $headers[(string) $item->Name] = (string) $item->Value->StringValue;
+
+                $headers[$name] = $attribute->getStringValue();
             }
+
             $this->buffer[] = [
-                'id' => (string) $xmlMessage->ReceiptHandle,
-                'body' => (string) $xmlMessage->Body,
+                'id' => $message->getReceiptHandle(),
+                'body' => $message->getBody(),
                 'headers' => $headers,
             ];
         }
 
         $this->currentResponse = null;
 
-        yield from $this->getPendingMessages();
+        return true;
     }
 
     public function setup(): void
     {
-        $parameters = [
-            'Action' => 'CreateQueue',
+        $this->configuration['auto_setup'] = false;
+        if ($this->client->queueExists([
             'QueueName' => $this->configuration['queue_name'],
-        ];
+            'QueueOwnerAWSAccountId' => $this->configuration['account'],
+        ])->isSuccess()) {
+            return;
+        }
+
+        if (null !== $this->configuration['account']) {
+            throw new InvalidArgumentException(sprintf('The Amazon SQS queue "%s" does not exists (or you don\'t have permissions on it), and can\'t be created when an account is provided.', $this->configuration['queue_name']));
+        }
+
+        $parameters = ['QueueName' => $this->configuration['queue_name']];
 
-        if ($this->isFifoQueue($this->configuration['queue_name'])) {
+        if (self::isFifoQueue($this->configuration['queue_name'])) {
             $parameters['FifoQueue'] = true;
         }
 
-        $this->call($this->configuration['endpoint'], $parameters);
+        $this->client->createQueue($parameters)->resolve();
```

Comment:
`resolve()` is not needed.

Reply:
I don't think we should remove this one.
If user don't have permission to create the queue, the exception will be triggered on object's destruct (after calls to queueExists bellow) which can be confusing for the user

Reply:
But the Result object is destructed immediately since we don't care about the return value for `createQueue()`.

Reply:
indeed... Removed (same for `delete` method)